apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/new/client.tsx

```diff
@@ -811,6 +814,54 @@ export const CreateKey: React.FC<Props> = ({ apiId, keyAuthId }) => {
                         ) : null}
                       </CardContent>
                     </Card>
+                    <Card>
+                      <CardContent className="justify-between w-full p-4 item-center">
+                        <div className="flex items-center justify-between w-full">
+                          <span>Recoverable</span>
+
+                          <FormField
+                            control={form.control}
+                            name="recoverEnabled"
+                            render={({ field }) => (
+                              <FormItem>
+                                <FormLabel className="sr-only">Recoverable</FormLabel>
+                                <FormControl>
+                                  <Switch
+                                    onCheckedChange={(e) => {
+                                      field.onChange(e);
+                                      if (field.value === false) {
+                                        resetLimited();
+                                      }
+                                    }}
+                                  />
+                                </FormControl>
+                              </FormItem>
+                            )}
+                          />
+                        </div>
+
+                        {form.watch("recoverEnabled") ? (
+                          <>
+                            {form.formState.errors.ratelimit && (
+                              <p className="text-xs text-center text-content-alert">
+                                {form.formState.errors.ratelimit.message}
+                              </p>
+                            )}
```

Comment:
I prefer if this was disabled or even hidden if the user does not have encryption enabled with a tooltip versus enabling it and then returning the error. Both should coexist e.g defensively handling it in the backend and disable it in the front it with a tooltip to explain how to get it enabled. This feature is a security trade off with high impact so let's make sure it is only clickable for people who have talked to Unkey.

Reply:
https://www.loom.com/share/394e9b93064146e1b6d04135689e996e?sid=4a7206b1-62d8-4e62-8931-046edc138c66

Disabled